rust/src/main.rs

```diff
@@ -0,0 +1,94 @@
+use std::{str::FromStr, sync::Arc};
+
+use solana_sdk::signature::Keypair;
+use thiserror::Error;
+use drift_sdk::{types::{Context, SdkError}, DriftClient, RpcAccountProvider};
+use anchor_client::Cluster;
+
+pub mod jit_proxy_client;
+pub use jit_proxy_client::JitProxyClient;
+
+use crate::jitter::{JitterStrategy, Shotgun, JitParams};
+pub mod jitter;
+use dotenv::dotenv;
+use std::env;
+
+pub type JitResult<T> = Result<T, JitError>;
+
+#[derive(Debug, Error)]
+pub enum JitError {
+    #[error("{0}")]
+    Drift(String),
+    #[error("{0}")]
+    Sdk(String),
+}
+
+
+impl From<drift::error::ErrorCode> for JitError {
+    fn from(error: drift::error::ErrorCode) -> Self {
+        JitError::Drift(error.to_string())
+    }
+}
+
+impl From<SdkError> for JitError {
+    fn from(error: SdkError) -> Self {
+        JitError::Sdk(error.to_string())
+    }
+}
+
+#[tokio::main]
+async fn main() {
+    env_logger::init();
+    dotenv().ok();
+
+    let api_key = env::var("RPC_KEY").expect("RPC_KEY must be set");
+    let private_key = env::var("PRIVATE_KEY").expect("PRIVATE_KEY must be set");
+
+    let pk_vec: Vec<u8> = private_key.trim_matches(|c| c == '[' || c == ']')
+        .split(',')
+        .map(|s| s.trim().parse::<u8>().expect("Failed to parse u8"))
+        .collect();
+
+    let pk_bytes: &[u8] = &pk_vec;
+    let rpc_url = format!("https://mainnet.helius-rpc.com?api-key={}", api_key);
```

Comment:
probably better to get entire RPC url from env var instead of locking users into helius